Client/Frontend/Sync/BraveCore/Bookmarks/BookmarkManager.swift

```diff
@@ -0,0 +1,303 @@
+// Copyright 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import Data
+import BraveCore
+import BraveShared
+import CoreData
+import Shared
+
+private let log = Logger.browserLogger
+
+class BookmarkManager: NSObject {
+
+    // MARK: Lifecycle
+    
+    init(bookmarksAPI: BraveBookmarksAPI?) {
```

Comment:
As mentioned above, making `BraveCoreMain` non-optional will allow this to be non-optional

Reply:
Same as https://github.com/brave/brave-ios/pull/4024#discussion_r707530930